README.md

```diff
@@ -38,6 +38,17 @@ jobs:
     steps:
       - name: checkout
         uses: actions/checkout@v3
+        
+      - name: set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: gradle
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
```

Comment:
This should be done in your repo, not on CI.
```
git update-index --chmod=+X gradlew
git add gradlew
git commit -m "Make gradlew executable"
```
This will make it work for anyone cloning your project on UNIX and Mac machines as well.

Reply:
Ok, agreed.

1. But since primary use of this emulator on the CI, doesnt it make sense to have it through CI?

2. Is JDK 11 line relevant to be put in the README in your opinion?

Thank you for the review.

Reply:
1. `gradlew` should be always committed with `x` flag set, CI or not. Gradle itself has not much to do with the CI, and even less with emulator/this GitHub Action, so I don't think setting up Gradle correctly belongs here. (see also next point)

2. Kind of, I would assume though that whoever adds this action already has a fully functioning CI that at the minimum runs `gradlew build` (`assemble` + `test` + `lint` and others). So every project would have a Java set up correctly for their AGP version. In addition to this action not being coupled with AGP at all (you can use Bazel, Buck as well), or even just run `adb shell am instrument` directly with a pre-built APK. So Java version might vary highly, even thought the most common case would be Java 11. Note that AGP 8 (currently in RC) will require Java 17 at runtime.

Actually thinking more of it, the [default Java in macos-latest](https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#java) could theoretically change at any time.

---

I think if you want to keep this PR, remove `cache: gradle`, remove `gradlew` setup and add the Java setup to all code blocks, not just one.

(Btw, I'm not a collaborator/maintainer, just a user)